.github/workflows/docker-build.yaml

```diff
@@ -0,0 +1,33 @@
+name: Docker Image Build
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+    workflow_dispatch: {}
+    push:
+        paths-ignore:
+            - '.infra/**'
+            - '.github/**'
+            - '.happy/**'
+            - 'dev_docs/**'
+            - 'example-dataset/**'
+        branches-ignore:
+          - main
+
+jobs:
+  argus_builder:
```

Comment:
Docker builds are done through this CI action. Once the image is built, the image tag is committed back to the PR. This tag will be used to deploy the stack.